docs: #1769 what's new page

micheleRP opened the pull request:
resolves #1769 

preview: https://deploy-preview-1791--redpanda-documentation.netlify.app/docs/get-started/whats-new/

micheleRP commented:
@bpraseed: Please see this small Cloud section in the What's New for 23.2 doc: https://deploy-preview-1791--redpanda-documentation.netlify.app/docs/get-started/whats-new/#repdanda-cloud. Is that messaging correct?

mattschumpert commented on docs/get-started/whats-new.mdx:
```diff
@@ -0,0 +1,89 @@
+
+## Follower fetching
+
+[Follower fetching](../../develop/consume-data/follower-fetching) lets a consumer fetch records from the closest replica of a topic partition, regardless of whether it's a leader or a follower. This can minimize costs and consumer read latency for clusters deployed across different data centers and availability zones.
```
'minimize cloud networking costs'

mattschumpert commented on docs/get-started/whats-new.mdx:
```diff
@@ -0,0 +1,89 @@
+
+## Kubernetes Operator
+
+Redpanda offers a new approach to deploying [Redpanda in Kubernetes](../../deploy/deployment-option/self-hosted/kubernetes/kubernetes-production-deployment/). You can choose to use Helm for its simplicity, or you can combine Helm with the Redpanda Operator to leverage custom resource definitions (CRDs) for a more GitOps-friendly deployment process. Redpanda Operator is the recommended deployment  ...
```
>  or you can combine Helm with the Redpanda Operator to leverage custom resource definitions (CRDs) for a more GitOps-friendly deployment process.

I'm not sure 'more GitOps friendly' is accurate. Helm is dev ops  friendly too.

@joejulian what would you say here.  'you can combine Helm with the Redpanda Operator to leverage additional automation for day 2 operations including self-healing, ....?'

joejulian replied:
Helm is dev-ops friendly, but gitops is where you check in your declarative state for deployment into your cluster through continuous release.

Since helm is an imperative step, it is not gitops friendly.

CRDs are a standard implementation detail. I feel like mentioning them here is superfluous.

```suggestion
Redpanda offers a new approach to deploying [Redpanda in Kubernetes](../../deploy/deployment-option/self-hosted/kubernetes/kubernetes-production-deployment/). You can choose to use Helm for its simplicity, or you can use the Redpanda Operator for a more GitOps-friendly declarative deployment process. Use either Helm or Operator as suits your needs, both are recommended.
```

micheleRP replied:
Thanks, Joe! Committing your suggestion.

mattschumpert commented on docs/get-started/whats-new.mdx:
```diff
@@ -0,0 +1,89 @@
+## New commands and properties
+
+- [`rpk profile`](../../reference/rpk/rpk-profile/): This allows users to switch between clusters seamlessly.
+- [`rpk cloud login`](../../reference/rpk/rpk-cloud/rpk-cloud-login/): This allows users to connect to cloud environments by browser.
```
connect to cloud environments using a browser.

mattschumpert commented on docs/get-started/whats-new.mdx:
```diff
@@ -0,0 +1,89 @@
+- [`rpk cloud login`](../../reference/rpk/rpk-cloud/rpk-cloud-login/): This allows users to connect to cloud environments by browser.
+- [`rpk topic trim-prefix`](../../reference/rpk/rpk-topic/rpk-topic-trim-prefix/): This allows users to delete records from a topic.
+- [`rpk topic describe-storage`](../../reference/rpk/rpk-topic/rpk-topic-describe-storage/): This provides detailed information about storage, including local and object disk usage and time of last write to object storage.
+- [Cluster support properties](../../reference/cluster-properties/#support-properties): Two new configuration flags for unsafe string checks.
```
@micheleRP 'support properties' seems to be the wrong link here (where are the unsafe string checks?). Also these can't be the only two new config properties overall (with all the other features)

micheleRP replied:
@mattschumpert I added [those 2 links](https://deploy-preview-1791--redpanda-documentation.netlify.app/docs/get-started/whats-new/#new-commands-and-properties), but I don't think we want to maintain a list of every new/changed command or property in this What's New. In the [Documentation improvements section](https://deploy-preview-1791--redpanda-documentation.netlify.app/docs/get-started/whats-new/#documentation-improvements), we're listing significant doc updates that users may not have noticed. Let me know if there are other properties you want added to this list.

mattschumpert commented on docs/get-started/whats-new.mdx:
```diff
@@ -0,0 +1,89 @@
+
+## Redpanda Console TLS termination
+
+To secure Redpanda Console using TLS, you can either let Redpanda Console handle TLS termination or you can offload it to an upstream component, such as a reverse proxy or a Cloud HTTPS LoadBalancer. [TLS termination](../../manage/security/console/tls-termination) refers to the process of decrypting incoming TLS-encrypted traffic. 
```
@weeco , @vladoschreiner  shouldn't we also document the new-ish AD/Keycload support for console, given it wasn't there in 23.1.1 IIRC?

mattschumpert commented on docs/get-started/whats-new.mdx:
```diff
@@ -0,0 +1,89 @@
+
+## Repdanda Cloud
+
+Redpanda [managed connectors](../../deploy/deployment-option/cloud/managed-connectors/) are fully-managed solutions for integrating different data systems with your real-time clusters in Redpanda Cloud. They remove the need for manually building integrations with the Kafka API or Kafka Connect, and they’re available in the Redpanda Console. 
```
Managed connectors isn't entirely new since 23.1.1 is it @vladoschreiner ? Should we be only highlighting the specific connectors we launched since April?

micheleRP replied:
managed connectors will go in What's New in Redpanda Cloud.

mattschumpert commented on docs/get-started/whats-new.mdx:
```diff
@@ -0,0 +1,89 @@
+Redpanda [managed connectors](../../deploy/deployment-option/cloud/managed-connectors/) are fully-managed solutions for integrating different data systems with your real-time clusters in Redpanda Cloud. They remove the need for manually building integrations with the Kafka API or Kafka Connect, and they’re available in the Redpanda Console. 
+
+:::note
+Redpanda Cloud releases new features on a continuous basis and uptakes Redpanda platform (self-hosted) versions. 
```
@micheleRP  how are we thinking about this "what's new" wrt cloud releases in general. Should we round up and include (but specifically clarify) this is what's new since April 2023 ?  Or something else

wdyt @bpraseed ?

bpraseed replied:
Cloud "what's new" is independent of core release. We will have a separate release announcement next month for that and also have requested @Feediver1 to have a separate doc section for cloud to capture that so we could do something similar to https://www.cockroachlabs.com/docs/releases/cloud or https://docs.snowflake.com/release-notes/new-features

micheleRP replied:
@mattschumpert Do you want to remove the What's New section for Redpanda Cloud, and keep a note at the bottom of the page that Redpanda Cloud releases on a continuous basis?

bpraseed commented:
> 

We should just highlight the delta of Connectors. Will let @vladoschreiner comment on this.

vladoschreiner commented:
> We should just highlight the delta of Connectors. Will let @vladoschreiner comment on this.

@bpraseed which release does this "What's new" refer to? Is it self-hosted 23.2 or general snapshot of new Redpanda's new features since 23.1 was released?

If 23.3:
- [KeyCloak](https://docs.redpanda.com/docs/manage/security/console/keycloak/) and [AD](https://docs.redpanda.com/docs/manage/security/console/azure-ad/) auth for Console
- [Console HTTPS termination](https://docs.redpanda.com/docs/manage/security/console/tls-termination/#use-redpanda-console-for-tls-termination)
- [Self-hosted connectors with MM2](https://github.com/redpanda-data/documentation/pull/1805)

If all new features since 23.1, including cloud:
- New Kafka connect setup & edit experience
- Connectors for Mongo ([import](https://docs.redpanda.com/docs/deploy/deployment-option/cloud/managed-connectors/create-mongodb-source-connector/), [export](https://docs.redpanda.com/docs/deploy/deployment-option/cloud/managed-connectors/create-mongodb-sink-connector/)), [MySQL/Debezium](https://docs.redpanda.com/docs/deploy/deployment-option/cloud/managed-connectors/create-mysql-source-connector/), [PSQL/Debezium](https://docs.redpanda.com/docs/deploy/deployment-option/cloud/managed-connectors/create-postgresql-connector/), JDBC ([import](https://docs.redpanda.com/docs/deploy/deployment-option/cloud/managed-connectors/create-jdbc-source-connector/), [export](https://docs.redpanda.com/docs/deploy/deployment-option/cloud/managed-connectors/create-jdbc-sink-connector/))
- [Grafana dashboard](https://github.com/redpanda-data/documentation/pull/1749) to monitor managed connectors
- [Connector log access](https://github.com/redpanda-data/documentation/issues/1641) for troubleshooting in the cloud

micheleRP commented:
> > We should just highlight the delta of Connectors. Will let @vladoschreiner comment on this.
> 
> @bpraseed which release does this "What's new" refer to? Is it self-hosted 23.2 or general snapshot of new Redpanda's new features since 23.1 was released?
> 
> If 23.3:
> 
> * [KeyCloak](https://docs.redpanda.com/docs/manage/security/console/keycloak/) and [AD](https://docs.redpanda.com/docs/manage/security/console/azure-ad/) auth for Console
> * [Console HTTPS termination](https://docs.redpanda.com/docs/manage/security/console/tls-termination/#use-redpanda-console-for-tls-termination)
> * [Self-hosted connectors with MM2](https://github.com/redpanda-data/documentation/pull/1805)
> 
> If all new features since 23.1, including cloud:
> 
> * New Kafka connect setup & edit experience
> * Connectors for Mongo ([import](https://docs.redpanda.com/docs/deploy/deployment-option/cloud/managed-connectors/create-mongodb-source-connector/), [export](https://docs.redpanda.com/docs/deploy/deployment-option/cloud/managed-connectors/create-mongodb-sink-connector/)), [MySQL/Debezium](https://docs.redpanda.com/docs/deploy/deployment-option/cloud/managed-connectors/create-mysql-source-connector/), [PSQL/Debezium](https://docs.redpanda.com/docs/deploy/deployment-option/cloud/managed-connectors/create-postgresql-connector/), JDBC ([import](https://docs.redpanda.com/docs/deploy/deployment-option/cloud/managed-connectors/create-jdbc-source-connector/), [export](https://docs.redpanda.com/docs/deploy/deployment-option/cloud/managed-connectors/create-jdbc-sink-connector/))
> * [Grafana dashboard](https://github.com/redpanda-data/documentation/pull/1749) to monitor managed connectors
> * [Connector log access](https://github.com/redpanda-data/documentation/issues/1641) for troubleshooting in the cloud

@mattschumpert, @bpraseed, @vladoschreiner, @Feediver1: I've removed the Cloud and Console sections from this file. This is the first release that we've included a What's New in our doc. I think going forward, this file should include links to any doc with meaningful customer updates since the last product release (so, next version will cover changes from 23.2 to 23.3). We'll need to have a separate file for Cloud. Please confirm and approve, if this is OK!

micheleRP commented on docs/get-started/whats-new.mdx:
```diff
@@ -0,0 +1,89 @@
+
+## Topic-aware leadership balancing
+
+The new [`leader_balancer_mode`](../../manage/cluster-maintenance/cluster-balancing/#partition-leadership-balancing) property ensures that each shard in a cluster has an equal number of partitions. It determines the movement of leadership for the replica set of a partition. As an alternative to Redpanda partition balancing, you can change partition assignments explicitly with the Kafka API or wit ...
```
@mattschumpert I added a line about improved data balancing, see https://deploy-preview-1791--redpanda-documentation.netlify.app/docs/get-started/whats-new/#topic-aware-leadership-balancing

micheleRP commented on docs/get-started/whats-new.mdx:
```diff
@@ -0,0 +1,89 @@
+
+As data in object storage grows, the metadata for it grows. To support efficient long-term data retention, Redpanda splits the metadata in object storage, storing metadata of only recently-updated segments, and safely archiving the remaining metadata in object storage and caching it locally on disk. Archived metadata can be loaded when historical data is accessed. This allows Tiered Storage to ha ...
+
+- [Fine-grained caching](../../manage/tiered-storage/#caching): To support more concurrent consumers of historical data with less local storage, Redpanda can download small chunks of remote segments from the cache directory. For example, when a client fetch request spans a subsection of a 1 GiB segment, instead of downloading the entire 1 GiB segment, Redpanda can download 16 MiB chunks that cont ...
```
Thanks, Evgeny! I've fixed that in the what's new and in the TS page.

mattschumpert commented on docs/get-started/whats-new.mdx:
```diff
@@ -0,0 +1,85 @@
+The following new features are included in version 23.2. For a complete list of updates, see the [Redpanda release notes](https://github.com/redpanda-data/redpanda/releases/tag/v23.2.1).
+
+:::note
+Redpanda Cloud releases new features on a continuous basis and uptakes Redpanda platform (self-hosted) versions. 
```
I suggest:

"Redpanda Cloud releases new features on a continuous basis and includes the latest Redpanda platform major version"

mattschumpert commented:
@micheleRP 

I think all the Console stuff  SINCE 23.1 SHOULD be included (this is the approach taken for Core as well, even if it came in a patch release this is our window to announce)

The Connectors content should NOT be included if this page is focused on the self-hosted product since we don't have self-manaed connectors yet.

micheleRP commented:
> @micheleRP
> 
> I think all the Console stuff SINCE 23.1 SHOULD be included (this is the approach taken for Core as well, even if it came in a patch release this is our window to announce)
> 
> The Connectors content should NOT be included if this page is focused on the self-hosted product since we don't have self-managed connectors yet.
> 
> Unless @vladoschreiner you consider the MM2 self-hosted connector GA (though I see it's doc ticket is still open)

I added a small section for Console enhancements: https://deploy-preview-1791--redpanda-documentation.netlify.app/docs/get-started/whats-new/#redpanda-console-enhancements. Please review and confirm @mattschumpert, @vladoschreiner, @Feediver1

Feediver1 commented on docs/get-started/whats-new.mdx:
```diff
@@ -0,0 +1,90 @@
+    <link rel="canonical" href="https://docs.redpanda.com/docs/get-started/whats-new/" />
+</head>
+
+The following new features are included in version 23.2. For a complete list of updates, see the [Redpanda release notes](https://github.com/redpanda-data/redpanda/releases/tag/v23.2.1).
```
After reading this, my first question is--what is the difference between this and Rns? If RNs are a complete list of updates, is this an incomplete list? I think the distinction is important/should be clear. 

How about: This topic includes new content added in version 23.2. For a complete list of all product updates, see..."

^^ This gives us more latitude for additions not specific to the version (I know you have a section for that here).

Thoughts?

micheleRP replied:
thanks, I've updated to this!

Feediver1 commented on docs/get-started/whats-new.mdx:
```diff
@@ -0,0 +1,90 @@
+
+As data in object storage grows, the metadata for it grows. To support efficient long-term data retention, Redpanda splits the metadata in object storage, maintaining metadata of only recently-updated segments in memory or local disk, while safely archiving the remaining metadata in object storage and caching it locally on disk. Archived metadata is then loaded only when historical data is access ...
+
+- [Fine-grained caching](../../manage/tiered-storage/#caching): To support more concurrent consumers of historical data with less local storage, Redpanda can download small chunks of remote segments to the cache directory. For example, when a client fetch request spans a subsection of a 1 GiB segment, instead of downloading the entire 1 GiB segment, Redpanda can download 16 MiB chunks that contai ...
```
Looking at the content in the link, and it's just called "caching"--where/why does the "fine-grained" come in?

micheleRP replied:
@mattschumpert @redpanda-data/cloud-ring0 the announcement blog calls these feature fine-grained caching and automatic disk space management. Our docs, which have been reviewed/approved, called these in our docs "caching" and "local storage housekeeping". Should we change the docs now to match this messaging? Or is the inconsistency OK?

Feediver1 replied:
@mattschumpert Consistency wrt nomenclature is paramount in our product docs. We cannot call these features two different things. I've asked @micheleRP to pick one name and stick with it throughout our docs. It appears that she had to cull through 7 different sources to get to this point/come up with the definitive list of features (another issue altogether). For now, please weigh in on a single name for these two features--that is what we will go with in the docs. Thanks!

mattschumpert replied:
Much of this is by design.  The context of 'what's new' (existing user upgrading) is slightly different from someone learning the product for the first time, so there is bound to be a little dissonance. 

1. Caching

 To an existing user trying to digest what's new it's 'more fine grained' than it used to be.  To a new user, they just want to understand how the TS cache works in general and they have no understanding of the previous behavior (where we cached by segment). , so this is 'caching'. Its absolutely correct that it's under the 'caching' docs topic in describing how the product works if learning it for the first time. It's also correct for out blogs and what's new to say 'fine-grained caching' or perhaps better 'finer grained caching'.

For this audience, let's go with 'finer-grained caching'.

2. On the disk space management , similar situation. We've always had local storage housekeeping (removing expired data), now housekeeping just functions in a way that's more automatic. It's the 'automatic' that's the new part.

However, to bridge the gap here, and as the new docs content covers a bit more than just the housekeeping process. let's rename the docs sub-section "l[ocal storage housekeeping](https://docs.redpanda.com/docs/manage/cluster-maintenance/disk-utilization/#local-storage-housekeeping)" to  just "Space management". It's already in a 'manage disk space' page which is by definition about local disk, and 'space management' more accurately covers this section and is more aligned.

micheleRP replied:
Thanks @mattschumpert, makes sense. I've updated the what's new page to use "finer-grained caching" and I've changed the "Local storage housekeeping" heading (in the Disk Management page) to "Space management".

Feediver1 commented on docs/get-started/whats-new.mdx:
```diff
@@ -0,0 +1,90 @@
+
+- [Fine-grained caching](../../manage/tiered-storage/#caching): To support more concurrent consumers of historical data with less local storage, Redpanda can download small chunks of remote segments to the cache directory. For example, when a client fetch request spans a subsection of a 1 GiB segment, instead of downloading the entire 1 GiB segment, Redpanda can download 16 MiB chunks that contai ...
+
+- [Automatic disk space management](../../manage/cluster-maintenance/disk-utilization/#local-storage-housekeeping): Redpanda now divides disk storage into different categories to provide a flexible configuration of space: reserved disk space, cache storage, and log storage. When data usage begins to approach the target size of log storage, Redpanda does local storage housekeeping to bring usage b ...
```
Same here--the terminology for the link does not match the terms used in the link destination.

Reason I'm asking about these is that consistency regarding what we call these bits is important. Is it fine-grained caching or just caching? Automatic disk space management, or just local storage housekeeping?

Feediver1 commented on docs/get-started/whats-new.mdx:
```diff
@@ -0,0 +1,90 @@
+
+## Delete records from a topic​
+
+Redpanda now lets you [delete data from the beginning of a partition up to a specific offset](../../develop/config-topics/#delete-records-from-a-topic). The offset represents the true creation time of the event, not the time when it was stored by Redpanda. Deleting records frees up space in local disk and in object storage, which is especially helpful if your producers are pushing more data than  ...
```
```suggestion
Redpanda enables you to [delete data from the beginning of a partition up to a specific offset](../../develop/config-topics/#delete-records-from-a-topic). The offset represents the true creation time of the event, not the time when it was stored by Redpanda. Deleting records frees up space in local disk and in object storage, which is especially helpful if your producers are pushing more data than you anticipated when sizing your storage infrastructure, or if you want to implement a data retention policy aligned with a particular business event and not based on age or size. There are different ways to delete records from a topic, including using the `rpk topic trim` command or using the DeleteRecords Kafka API with Kafka clients.
```

vladoschreiner commented:
> Unless @vladoschreiner you consider the MM2 self-hosted connector GA (though I see it's doc ticket is still open)

The feature is ready. We're waiting for docs. My guidance on what to include in What's new:
- Include the self-hosted docker image. Prioritize merging the docs. The feature is ready and the docs are just waiting for a review. https://github.com/redpanda-data/documentation/pull/1805
- Don't include self-hosted MM2. The docs are missing. https://github.com/redpanda-data/documentation/issues/1642

Feediver1 reviewed (approved):
Thanks for this @micheleRP

micheleRP commented:
> > Unless @vladoschreiner you consider the MM2 self-hosted connector GA (though I see it's doc ticket is still open)
> 
> The feature is ready. We're waiting for docs. My guidance on what to include in What's new:
> 
> * Include the self-hosted docker image. Prioritize merging the docs. The feature is ready and the docs are just waiting for a review. [Docs: added self-managed connectors docker image documentation #1805](https://github.com/redpanda-data/documentation/pull/1805)
> * Don't include self-hosted MM2. The docs are missing. [Self-managed mirrormaker2 connector #1642](https://github.com/redpanda-data/documentation/issues/1642)

Added connector Docker image: https://deploy-preview-1791--redpanda-documentation.netlify.app/docs/get-started/whats-new/#redpanda-console-enhancements

Feediver1 commented on modules/deploy/partials/requirements.adoc:
```diff
@@ -88,7 +88,7 @@ endif::[]
 *Requirements*:
 
-- Each node must have at least two physical CPU cores.
+- Each production node must have at least four physical CPU cores.
```
@tmgstevens This update contradicts:
**[CPU and memory](https://docs.redpanda.com/streaming/current/deploy/redpanda/manual/production/requirements/#cpu-and-memory)**
Requirements:

Each node must have at least two physical CPU cores.

tmgstevens replied:
Let's leave this as 2 per the rest of the docs

Feediver1 commented on modules/deploy/partials/requirements.adoc:
```diff
@@ -88,7 +88,7 @@ endif::[]
 *Requirements*:
 
-- Each node must have at least two physical CPU cores.
+- Each production node must have at least four physical CPU cores.
```
```suggestion
- Each production node must have at least two physical CPU cores.
```
